### PR TITLE
deps: update kube to 0.85.0 and k8s-openapi to 0.19.0+v1_25

### PR DESCRIPTION
**Pull request by rjobanp**
### Motivation

To avoid duplicate package version warnings on `serde_yaml` in the cloud repo caused by the introduction of `utoipa` for generating OpenAPI specs, we need to update the `kube` version to 0.85.0. However doing so in the cloud repo (and in https://github.com/MaterializeInc/k8s-controller) requires doing so in the materialize repo first, hence this PR.

### Tips for reviewer

The one code change is due to a slight type change in the kube crate: https://github.com/kube-rs/kube/blob/main/CHANGELOG.md#listwatch-changes

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by nrainer-materialize** [comment]
I triggered a nightly build: https://buildkite.com/materialize/nightlies/builds/3104

**Comment by rjobanp** [comment]
Still have a number of crates version conflicts to resolve so I'm not sure if it's worth doing so yet

**Comment by nrainer-materialize** [comment]
> Still have a number of crates version conflicts to resolve so I'm not sure if it's worth doing so yet

Ok, I have cancelled the build. Though, we should run one on the final state before the merge. 😀

**Comment by rjobanp** [comment]
Alright I have cargo `clippy` and cargo `deny` passing by allowlisting the remaining duplicates in `deny.toml` https://github.com/MaterializeInc/materialize/pull/21034/files#diff-1040309c64844eb1b6b63d8fd67938adbf9461f1b3c61f12cf738f064a02d3de

I left a few TODOs in that file for the ones that I think can be resolved by creating new forks and modifying some intermediate deps, however it doesn't seem that I have permissions to create new repos in our github org, so would need some help there (@benesch ?)

**Comment by nrainer-materialize** [comment]
> Is this normal flakiness or something potentially caused by a crate upgrade? I'm unfamiliar with this test-suite

I triggered a retry but am not aware of these tests flaking recently. I also couldn't find a [CI-flake issue](https://github.com/search?q=repo%3AMaterializeInc%2Fmaterialize+%22checks+%2B%22+label%3Aci-flake+state%3Aopen+&type=issues) and the [tests pipeline also doesn't show other occurrences.](https://buildkite.com/materialize/tests/builds?branch=main)

Is this in line with your observations, @philip-stoev?

**Comment by philip-stoev** [comment]
Yes, this particular failure has not been seen previously. You can try running the statements manually in a `psql` client to see if they fail for you .

**Comment by rjobanp** [comment]
Yeah there doesn't seem to be.. this may be a bit of a dead end. Spending a few more cycles seeing if I can get tonic at `0.8.3` to still avoid a whole bunch of new conflicts, but if I can't figure that out soon I may have to punt on this whole effort.

**Comment by benesch** [comment]
Don't abandon ship yet! We can get this working without trouble. Good find on the limit issue.

**Comment by benesch** [comment]
The main GRPC server is configured here:

https://github.com/MaterializeInc/materialize/blob/1127cf9fd6158b5258565d2a7ca948373ad7e52d/src/service/src/grpc.rs#L282-L286

The only other place that creates a gRPC server that I'm aware of is the persist PubSub client:

https://github.com/MaterializeInc/materialize/blob/1127cf9fd6158b5258565d2a7ca948373ad7e52d/src/persist-client/src/rpc.rs#L944C10-L949

You'll need to ask the persist team (@MaterializeInc/persist) about what the default should be there.

**Comment by rjobanp** [comment]
I pushed a commit setting the decoding size for the gRPC servers but the test is failing on the mz-compute client and I haven't figured out where to insert the same config for that yet (the `compute-client` defined here https://github.com/MaterializeInc/materialize/blob/roshan/kube-0.85/src/compute-client/src/service.rs#L66-L67 as an example) https://docs.rs/tonic/latest/tonic/client/struct.Grpc.html#method.max_decoding_message_size

**Comment by rjobanp** [comment]
Alright the latest commit applies the limits for gRPC clients too, and the tests pass locally now. I've re-triggered the nightly test on this branch too https://buildkite.com/materialize/nightlies/builds/3138#0189d6bf-f682-485a-b0b4-8dc57d00ab1f

**Comment by rjobanp** [comment]
Okay the newest failure is due to the patch brought in from https://github.com/MaterializeInc/kube-rs/ -- I need help to set the default branch on that repo to `base64-update` instead of `fix_duplicate_finalizers`, or to merge `base64-update` to `main` and set that as the default.

The default branch we have set on that repo is from mid-2022 https://github.com/MaterializeInc/kube-rs/tree/fix_duplicate_finalizers that I assume was an older patch we used but is no longer in-use.

I created a branch called `base64-update` https://github.com/MaterializeInc/kube-rs/tree/base64-update to fix the base64 dependency duplicate but it looks like our `unused-dependencies` test is failing since it just clones the patch repo default branch and then can't find the reference patch commit object locally.

**Comment by necaris** [comment]
@rjobanp I think we can specify the `branch =` parameter to the git dependency? https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories

**Comment by rjobanp** [comment]
@necaris I did try that locally, but for some reason it would also pull back in the patch from the default-branch as a secondary `0.74` version in Cargo.lock and then complain about duplicate versions again, despite nothing referencing that `0.74` version. So I think there is some sort of bug or weird issue with the way the patch cloning works and the default git branch on the remote

**Comment by benesch** [comment]
Oh, hm, kube-rs was using a nonstandard strategy for one of our forks. I've updated it to look more like the rest. `master` is now pointing at the commit you had on the `base64-update` branch. I worked some Git magic to graft your change onto a history that also includes the commits on the `fix_duplicate_finalizers` branch. That way the `master` branch preserves all prior commits that were previously referenced by Materialize. I also made `master` the default branch.

So tl;dr hopefully this is fixed now! 🤞🏽

**Comment by rjobanp** [comment]
Thank you @benesch ! The let it work locally so I just pushed a new commit and I'm fairly confident this PR should be good to merge now. Just triggered the nightlies again. Would appreciate a review from anyone who feels comfortable doing so!

File: src/orchestrator-kubernetes/src/lib.rs [hunk 1]
```diff
@@ -353,12 +353,12 @@ impl k8s_openapi::Metadata for MetricValueList {
...
     /// Return a `ListParams` instance that limits results to the namespace
     /// assigned to this orchestrator.
-    fn list_pod_params(&self) -> ListParams {
+    fn list_pod_params(&self) -> watcher::Config {
         let ns_selector = format!(
             "environmentd.materialize.cloud/namespace={}",
             self.namespace
         );
-        ListParams::default().labels(&ns_selector)
+        watcher::Config::default().labels(&ns_selector)
```
Comment (necaris) [inline src/orchestrator-kubernetes/src/lib.rs:361]: Should we just rename this method, since it looks like it's only used in a `watch()` invocation anyway?

File: src/persist-client/src/rpc.rs [hunk 2]
```diff
@@ -943,7 +943,7 @@ impl PersistGrpcPubSubServer {
     /// Starts the gRPC server. Consumes `self` and runs until the task is cancelled.
     pub async fn serve(self, listen_addr: SocketAddr) -> Result<(), anyhow::Error> {
         tonic::transport::Server::builder()
-            .add_service(ProtoPersistPubSubServer::new(self))
+            .add_service(ProtoPersistPubSubServer::new(self).max_decoding_message_size(usize::MAX))
```
Comment (necaris) [inline src/persist-client/src/rpc.rs:946]: Should we add a comment here about the use of `usize::MAX`?

**Review by necaris — APPROVED** [review]
A couple of nits but 🚀 thank you @rjobanp for taking on all this yak-shaving!

File: src/persist-client/src/rpc.rs (hunk 2 above)
Comment (rjobanp) [reply src/persist-client/src/rpc.rs:946]: I'll add a comment, the reason I didn't use `MAX_GRPC_MESSAGE_SIZE` is that this crate doesn't rely on the  `service/grpc` library in the same way that the other servers do.
